app/angular/src/server/angular-cli_config.js

```diff
@@ -0,0 +1,69 @@
+const path = require('path');
+const fs = require('fs');
+
+export function getAngularCliWebpackConfigOptions(dirToSearch, appIndex = 0) {
+  const fname = path.join(dirToSearch, '.angular-cli.json');
+  if (!fs.existsSync(fname)) {
+    return null;
+  }
+  const cliConfig = JSON.parse(fs.readFileSync(fname, 'utf8'));
+  if (!cliConfig.apps || !cliConfig.apps.length) {
+    throw new Error('.angular-cli.json must have apps entry.');
+  }
+  const appConfig = cliConfig.apps[appIndex];
+
+  // FIXME dummy value
+  const cliWebpackConfigOptions = {
+    projectRoot: dirToSearch,
+    appConfig,
+    buildOptions: {
+      outputPath: 'outputPath',
+    },
+    supportES2015: false,
+  };
+
+  return cliWebpackConfigOptions;
+}
+
+export function applyAngularCliWebpackConfig(baseConfig, cliWebpackConfigOptions) {
+  if (!cliWebpackConfigOptions) return baseConfig;
+
+  let ngcliConfigFactory;
+  try {
+    // We should not require('@angular/cli') at the top script level because user project might not have `@angular/cli`.
+    // eslint-disable-next-line global-require, import/no-extraneous-dependencies
+    ngcliConfigFactory = require('@angular/cli/models/webpack-configs');
+  } catch (e) {
```

Comment:
I'd like to see this changed a bit.

I'd like to extract a function that detects the installment of `@angular/cli`.

If it's not installed, we just merge the base, we might log that.
If it's installed we try to get their webpack config.
This might fail, since it's a deep file in a npm module.
We should log in either case.